include/wil/win32_helpers.h

```diff
@@ -317,7 +317,7 @@ namespace wil
                 // bufferLength includes the null so subtract that as make() will add space for it.
                 RETURN_IF_FAILED(maker.make(nullptr, bufferLength - 1));
 
-                RETURN_IF_FAILED_EXPECTED(callback(maker.buffer(), bufferLength, &valueLengthNeededWithNull));
+                RETURN_IF_FAILED_EXPECTED(callback(const_cast<wchar_t*>(maker.buffer()), bufferLength, &valueLengthNeededWithNull));

```

Comment:
`const_cast` should not be needed. If something changed to require it, that should be reverted